src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/executors/spark/SparkClusterService.scala

```diff
@@ -32,10 +32,20 @@ trait SparkClusterService {
   protected def mergeAdditionalSparkConfig(
     globalConfig: Map[String, String],
     jobConfig: Map[String, String]
-  ): Map[String, String] =
-    KeysToMerge.map { key =>
+  ): Map[String, String] = {
+    val extraJavaOptionsMerge = SparkExtraJavaOptions.KeysToMerge.map { key =>
       val globalValue = globalConfig.getOrElse(key, "")
       val jobValue = jobConfig.getOrElse(key, "")
-      key -> s"$globalValue$MergedValuesSeparator$jobValue".trim
-    }.toMap
+      key -> s"$globalValue${SparkExtraJavaOptions.MergedValuesSeparator}$jobValue".trim
+    }
+    val tagsOptions = SparkTags.KeysToMerge.map { key =>
+      val globalValue = globalConfig.get(key)
+      val jobValue = jobConfig.get(key)
```

Comment:
Add some test code, or try to reuse the logic in `JobTemplateResolutionService`. For example, you could use `.getOrElse(Array.empty)` here.

Reply:
I added tests.
I cannot use .getOrElse(Array.empty) because, I need a string.
This gives me Option which I can filter out if is None. 
I could use .getOrElse("") but this has to be filtered anyway because if I split "" it gives me an array with one empty element, so I would rather have Optional rather then String and EmptyString